### PR TITLE
Check if field is present in document when aggregating.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is fixing an issue with the scripted terms aggregation we are generating. When multiple indices are involved, one (or more) might lack the field mapping for one (or more) of the pivot fields involved. This results in an exception like this one:

```
Elasticsearch exception [type=illegal_argument_exception, reason=No field found for [took_ms_int] in mapping with types []]
```

This PR is now adding a check (`doc.containsKey('field')`) for the field presence in the current document to avoid this.

/nocl
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/buckets/ESValuesHandler.java [hunk 1]
```diff
@@ -105,11 +105,26 @@ private TermsAggregationBuilder createScriptedTerms(List<? extends BucketSpec> b
 
     private Script scriptForPivots(Collection<? extends BucketSpec> pivots) {
         final String scriptSource = Joiner.on(KEY_SEPARATOR_PHRASE).join(pivots.stream()
-                .map(bucket -> "String.valueOf(doc['" + bucket.field() + "'].size() == 0 ? \"N/A\" : doc['" + bucket.field() + "'].value)")
+                .map(bucket -> "String.valueOf((" + documentHasField(bucket.field())
```
Comment (todvora) [inline graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/buckets/ESValuesHandler.java:108]: What about a multiline string interpolation? Wouldn't it increase readability?

Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/buckets/ESValuesHandler.java]: Good idea. Did that!

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/buckets/ESValuesHandler.java [hunk 2]
```diff
@@ -105,7 +105,9 @@ private TermsAggregationBuilder createScriptedTerms(List<? extends BucketSpec> b
 
     private Script scriptForPivots(Collection<? extends BucketSpec> pivots) {
         final String scriptSource = Joiner.on(KEY_SEPARATOR_PHRASE).join(pivots.stream()
-                .map(bucket -> "String.valueOf(doc['" + bucket.field() + "'].size() == 0 ? \"N/A\" : doc['" + bucket.field() + "'].value)")
+                .map(bucket -> """
+                        String.valueOf((doc.containsKey('%s') && doc['%s'].size() > 0) ? doc['%s'].value : "%s")
```
Comment (luk-kaminski) [inline graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/buckets/ESValuesHandler.java:109]: What about "%1$s" (so called "Argument Index") format string and using "bucket.field()" only once?

File: graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/views/searchtypes/pivot/buckets/OSValuesHandler.java [hunk 3]
```diff
@@ -126,7 +126,9 @@ private TermsAggregationBuilder createScriptedTerms(List<? extends BucketSpec> b
 
     private Script scriptForPivots(Collection<? extends BucketSpec> pivots) {
         final String scriptSource = Joiner.on(KEY_SEPARATOR_PHRASE).join(pivots.stream()
-                .map(bucket -> "String.valueOf(doc['" + bucket.field() + "'].size() == 0 ? \"" + MissingBucketConstants.MISSING_BUCKET_NAME + "\" : doc['" + bucket.field() + "'].value)")
+                .map(bucket -> """
+                        String.valueOf((doc.containsKey('%s') && doc['%s'].size() > 0) ? doc['%s'].value : "%s")
+                        """.formatted(bucket.field(), bucket.field(), bucket.field(), MissingBucketConstants.MISSING_BUCKET_NAME))
```
Comment (janheise) [inline graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/views/searchtypes/pivot/buckets/OSValuesHandler.java:131]: is the order of the parameters correct? MISSING_BUCKET at the end while it's not the last parameter?

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/buckets/ESValuesHandler.java (hunk 2 above)
Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/searchtypes/pivot/buckets/ESValuesHandler.java:109]: Oh great, I was looking for something like this! :pray:

**Pull request merged**